python/external_config/scripts/external_runner.py

```diff
@@ -143,6 +152,44 @@ def execute_command(self):
                 self.completed.emit()
 
 
+def _handle_qt_warnings():
+    """
+    This will suppress the libpng warnings, but allow any
+    other warnings or errors from QT to print.
+    We do this because warning by default get printed to stderr
+    and so they get displayed in the browser when running actions.
+    """
+
+    def handler(*args):
+        # We handle the args this way since Qt 5 passes 3 args, type, context and message,
+        # Where as Qt 4 only passes 2, type, and message.
+        msg_type = args[0]
+        msg_string = args[-1]
+
+        # Suppress this warning.
+        if msg_string == "libpng warning: iCCP: known incorrect sRGB profile":
+            return
+
+        if msg_type in [
+            qt_importer.QtCore.QtMsgType.QtWarningMsg,
+            qt_importer.QtCore.QtMsgType.QtCriticalMsg,
+            qt_importer.QtCore.QtMsgType.QtFatalMsg,
+        ]:
+            # By default Qt would usually print these to stderr so we should do the same.
+            print(msg_string, file=sys.stderr)
+        else:
+            # This is probably a debug or info message so just print these normally.
+            print(msg_string)
+
+    # Add a message handler so we can suppress the warnings about libpng.
+    try:
+        # QT 4
+        qt_importer.QtCore.qInstallMsgHandler(handler)
+    except AttributeError:
+        # QT 5
+        qt_importer.QtCore.qInstallMessageHandler(handler)
```

Comment:
Maybe this stuff should be moved into core as a utility method? could be useful?

Reply:
Let's wait and see if this becomes a pattern first.